app/controllers/katello/products_controller.rb

```diff
@@ -60,7 +60,7 @@ def find_content
     end
 
     def substitutions
-      params.slice(:basearch, :releasever, :registry_name)
+      params.permit(:basearch, :releasever, :registry_name).to_h
```

Comment:
why is htis needed exactly?  doesn't permit return something that is usable as a hash?

Reply:
High level: enabling/disabling repos was broken.

.permit gives us an instance of ActionController::Parameters which I perceive as the correct way to pass params outside the controller. However, passing such an instance deeper gets us into trouble and it's fixed by sending the exected hash.

This happens w/o to_h:

```
2018-01-05T15:07:32 1451fc76 [background] [E] undefined method `reduce' for nil:NilClass (NoMethodError)                                                                                                           
 | /home/vagrant/katello/app/lib/katello/util/path_with_substitutions.rb:46:in `apply_substitutions'                                                                                                               
 | /home/vagrant/katello/app/lib/katello/util/cdn_var_substitutor.rb:26:in `validate_substitutions'                                                                                                                
 | /home/vagrant/katello/app/models/katello/candlepin/repository_mapper.rb:62:in `validate!'                                                                                                                       
 | /home/vagrant/katello/app/lib/actions/katello/repository_set/enable_repository.rb:11:in `plan'
```

Reply:
![screenshot from 2018-01-05 18-00-39](https://user-images.githubusercontent.com/598891/34619504-a64a4916-f242-11e7-8e2e-104a023261b1.png)